### DISCUSSION TITLE
Custom formatter in external cop extension

### DISCUSSION THREAD

**Discussion by renjiexu** [rubocop/rubocop#9697] [Q&A] [answered]
Hi team,

I'm building a new cop as a separate gem, meanwhile I'm planning to have a custom formatter just for the this cop, however, I couldn't figure out a way to specify the formatter in gem, the docs is using `--require` parameter: https://docs.rubocop.org/rubocop/extensions.html#using-a-custom-formatter-from-the-command-line, but I don't want to keep a copy of formatter file in each repos that using this new gem, wondering if there is a way to achieve that? Thanks!

> bundle exec rubocop app/controllers/ping_controller.rb --format MagicFormatter  --only CustomCop/CheckMagic

```
uninitialized constant MagicFormatter
/usr/local/bundle/ruby/2.5.0/gems/rubocop-1.12.1/lib/rubocop/formatter/formatter_set.rb:103:in `block in custom_formatter_class'
/usr/local/bundle/ruby/2.5.0/gems/rubocop-1.12.1/lib/rubocop/formatter/formatter_set.rb:102:in `each'
/usr/local/bundle/ruby/2.5.0/gems/rubocop-1.12.1/lib/rubocop/formatter/formatter_set.rb:102:in `reduce'
/usr/local/bundle/ruby/2.5.0/gems/rubocop-1.12.1/lib/rubocop/formatter/formatter_set.rb:102:in `custom_formatter_class'
/usr/local/bundle/ruby/2.5.0/gems/rubocop-1.12.1/lib/rubocop/formatter/formatter_set.rb:80:in `formatter_class'
```

**Comment by koic** [comment, marked as answer]
It is possible to specify gem name instead of relative path. Alternatively, `require` can be specified in .rubocop.yml as shown below.

```ruby
# .rubocop.yml
require:
  - ../my/custom/file.rb
  - rubocop-extension
```

- https://docs.rubocop.org/rubocop/extensions.html#loading-extensions
- https://github.com/rubocop/rubocop/blob/v1.12.1/.rubocop.yml#L4-L7

How to use rubocop-performance may be helpful.
https://github.com/rubocop/rubocop-performance#usage

**Reply by renjiexu** [reply]
ah, you're right, able to load the formatter by specifying the gem in the require param:

```
bundle exec rubocop app/controllers/ping_controller.rb --format MagicFormatter --only CustomCop/CheckMagic --require rubocop-magic
``` 

thanks!
